spring-kafka/src/main/java/org/springframework/kafka/listener/ConcurrentMessageListenerContainer.java

```diff
@@ -230,6 +234,19 @@ private void configureChildContainer(int index, KafkaMessageListenerContainer<K,
 			stopAbnormally(() -> {
 			});
 		});
+		AsyncListenableTaskExecutor exec = container.getContainerProperties().getConsumerTaskExecutor();
+		if (exec == null) {
+			if ((this.executors.size() > index)) {
+				exec = this.executors.get(index);
+			}
+			else {
+				String containerBeanName = container.getBeanName();
+				exec = new SimpleAsyncTaskExecutor(
+						(containerBeanName == null ? "" : containerBeanName) + "-C-");
```

Comment:
It cannot be `null`. See logic in the beginning of this method: `container.setBeanName((beanName != null ? beanName : "consumer") + "-" + index);`